roles/lib_openshift/src/lib/base.py

```diff
@@ -243,7 +266,10 @@ def openshift_cmd(self, cmd, oadm=False, output=False, output_type='json', input
         if self.verbose:
             print(' '.join(cmds))
 
-        returncode, stdout, stderr = self._run(cmds, input_data)
+        try:
+            returncode, stdout, stderr = self._run(cmds, input_data)
```

Comment:
> - No need for a locate_oc_binary function -- it is too specific, it seems what we want is to update $PATH and let Python/ the shell do its default operation.

@rhcarvalho that was discussed previously and with Ansible using both multi-process and threading, we decided to avoid potential issues with manipulating the env: https://github.com/openshift/openshift-ansible/pull/3413#issuecomment-280856499

> - Change _run to update the PATH, just like we set KUBECONFIG (curr_env.update({'KUBECONFIG': self.kubeconfig}))

We've discussed moving away from manipulating the env for KUBECONFIG as well, and moving towards the command line option instead, because of the discussion linked above.

> And if we're touching _run... there's usually a simpler way than using subprocess.Popen: https://docs.python.org/2/library/subprocess.html#using-the-subprocess-module (see call, check_call and check_output).

+1, though I've also wondered if we should move to something like [sh](https://amoffat.github.io/sh/). It would need to be packaged for RHEL, since it is only in Fedora/EPEL currently.